### DISCUSSION TITLE
Energy dashboard bar chart improvement

### DISCUSSION THREAD

**Discussion by j-kwan** [home-assistant/frontend#23152] [Dashboard]
Hi,

I would like to propose an improvement of the display of the Energy Dashboard.
Currently, the bar chart is displaying in the current order (top to bottom) :

Above 0w
1. Energy consumed from battery
2. Energy consumed from grid
3. Energy consumed from solar panel

Below 0w :
5. Energy discharged to the grid
6. Energy charging the battery

Is it possible to re-order it in this way ? 

Above 0w :
1. Energy consumed from grid
2 Energy consumed from battery
3. Energy consumed from solar panel

Below 0w :
5. Energy charging the battery
6. Energy discharged to the grid

In this way, the energy data nearer the 0w line are the energy for self-consumption and the external source are the farthest of the 0w line.
The order of the consumption will be the same as most of the self-consumption station : first from solar panel, then battery, and grid if the battery cannot provide the needed energy. Same thing for the energy storage : first charging battery and if there is a surplus, discharge to the grid

Best regards

**Comment by foyaxe** [comment]
This is exactly what I was thinking since I have my ESS and wanted to propose today.
It feels more intuitive to have the battery on the inside and the grid on the outside. For ESS users, the grid energy is what 'exceeds' the battery charging/discharging  power or capacity.
